src/lava/magma/core/model/py/model.py

```diff
@@ -29,8 +31,9 @@ class AbstractPyProcessModel(AbstractProcessModel, ABC):
         du: int =          LavaPyType(int, np.uint16, precision=12)
     """
 
-    def __init__(self, proc_params: ty.Dict[str, ty.Any]) -> None:
-        super().__init__(proc_params)
+    def __init__(self, proc_params: ty.Dict[str, ty.Any],
+                 loglevel=logging.WARNING) -> None:
+        super().__init__(proc_params, loglevel=loglevel)
```

Comment:
If you enable logging also for PyProcModels then I would also expect to see any code changes in the compiler, builder or Runtime to pass their 'loglevel' on to the ProcModel because the user won't instantiate the ProcModel directly.
Either the compiler passes log level to builder and Builder.build() passes it to the ProcModel

Or the compiler only uses it's loglevel for itself and the Runtime passes its loglevel to the builders at the actual runtime which passes it on to the ProcModels that are being built.

I believe the latter may make more sense because you will not always run a ProcModel at the same time you compile it. Therefore it makes more sense to derive the ProcModel loglevel from the Runtime.

Reply:
We definitely want to trickle down the loglevel from process run through all of the components.

From a new understanding of logging module I think we can set a loglevel at the root logger that will trickle down to other loggers. I'm testing locally now.